client/src/components/utils/Login/index.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import logInSchema from '../../../validation/login';
+import Button from '../Button/index';
+import background from '../../../assets/images/login.background.png';
+import waiter from '../../../assets/images/waiter.png';
+import './style.css';
+
+export default class Login extends React.Component {
+  state = {
+    tableNumber: null,
+    secret: '',
+    err: null,
+  };
+
+  setTableNumber = e => {
+    this.setState({ tableNumber: e.target.value });
+  };
+
+  setSecret = e => {
+    this.setState({ secret: e.target.value });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const { tableNumber, secret } = this.state;
+    const { history, updateTableNumber } = this.props;
+    logInSchema
+      .validateAsync({ tableNumber, secret })
+      .then(() => {
+        return fetch('/api/v1/login', {
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify({
+            tableNumber,
+            secret,
+          }),
+        });
+      })
+      .then(res => {
+        if (res.status === 200) {
+          updateTableNumber(tableNumber);
```

Comment:
`updateTableNumber` calls `setState` which means the process of updating the table number in the `App` component is `async` which may cause a problem when there is trouble updating the table number in the `App` component as you technically redirecting the user before updating the state value in the app.

### Solution
refactor `updateStateNumber` to take a callback and call `history.push` in the callback.